src/measure_theory/l2_space.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2021 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import analysis.normed_space.inner_product
+import measure_theory.set_integral
+
+/-! # `L^2` space
+
+-/
+
+noncomputable theory
+open topological_space measure_theory measure_theory.Lp
+open_locale nnreal ennreal
+
+namespace measure_theory
+
+variables {α E F G 𝕜 : Type*} [is_R_or_C 𝕜] [measurable_space α] {μ : measure α}
+  [measurable_space E] [inner_product_space 𝕜 E] [borel_space E] [second_countable_topology E]
+  [normed_group F] [measurable_space F] [borel_space F] [second_countable_topology F]
+
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 E _ x y
+
+lemma two_mul_le_add_sq (a b : ℝ) : 2 * a * b ≤ a ^ 2 + b ^ 2 :=
+begin
+  suffices h_nonneg : 0 ≤ a ^ 2 + b ^ 2 - 2 * a * b, by rwa sub_nonneg at h_nonneg,
+  calc 0 ≤ (a - b) ^ 2               : pow_two_nonneg _
+     ... = a ^ 2 + b ^ 2 - 2 * a * b : by ring,
+end
+
+lemma snorm_rpow_two_norm_lt_top (f : Lp F 2 μ) :
+  snorm (λ (x : α), ∥f x∥ ^ (2 : ℝ)) 1 μ < ∞ :=
+begin
+  have h_two : ennreal.of_real (2 : ℝ) = 2, by simp [zero_le_one],
+  rw [snorm_norm_rpow f zero_lt_two, one_mul, h_two],
+  exact ennreal.rpow_lt_top_of_nonneg zero_le_two (Lp.snorm_ne_top f),
+end
+
+lemma snorm_inner_lt_top (f g : Lp E 2 μ) : snorm (λ (x : α), ⟪f x, g x⟫) 1 μ < ∞ :=
+begin
+  have h : ∀ x, is_R_or_C.abs ⟪f x, g x⟫ ≤ ∥f x∥ * ∥g x∥,
+    from λ x, abs_inner_le_norm _ _,
+  have h' : ∀ x, is_R_or_C.abs ⟪f x, g x⟫ ≤ ∥ ∥f x∥^2 + ∥g x∥^2 ∥,
+  { simp_rw real.norm_eq_abs,
+    refine λ x, le_trans (h x) _,
+    rw abs_eq_self.mpr,
+    swap, { exact add_nonneg (by simp) (by simp), },
+    refine le_trans _ (half_le_self (add_nonneg (pow_two_nonneg _) (pow_two_nonneg _))),
+    refine (le_div_iff (@zero_lt_two ℝ _ _)).mpr (le_trans (le_of_eq _) (two_mul_le_add_sq _ _)),
+    ring, },
+  simp_rw [← is_R_or_C.norm_eq_abs, ← real.rpow_nat_cast] at h',
+  refine (snorm_mono_ae (ae_of_all _ h')).trans_lt ((snorm_add_le _ _ le_rfl).trans_lt _),
+  { exact (Lp.ae_measurable f).norm.rpow_const, },
+  { exact (Lp.ae_measurable g).norm.rpow_const, },
+  simp only [nat.cast_bit0, ennreal.add_lt_top, nat.cast_one],
+  exact ⟨snorm_rpow_two_norm_lt_top f, snorm_rpow_two_norm_lt_top g⟩,
+end
+
+section inner_product_space
+
+variables [measurable_space 𝕜] [borel_space 𝕜]
+
+include 𝕜
+
+instance : has_inner 𝕜 (Lp E 2 μ) := ⟨λ f g, ∫ a, ⟪f a, g a⟫ ∂μ⟩
+
+lemma inner_def (f g : Lp E 2 μ) : inner f g = ∫ a : α, ⟪f a, g a⟫ ∂μ := rfl
+
+lemma integral_inner_eq_sq_snorm (f : Lp E 2 μ) :
+  ∫ a, (inner (f a) (f a) : 𝕜) ∂μ = ennreal.to_real ∫⁻ a, (nnnorm (f a) : ℝ≥0∞) ^ (2:ℝ) ∂μ :=
```

Comment:
```suggestion
 ∫ a, ⟪f a, f a⟫ ∂μ = ennreal.to_real ∫⁻ a, (nnnorm (f a) : ℝ≥0∞) ^ (2:ℝ) ∂μ :=
```
Thanks for switching to use the notation!  One missed opportunity here.

Reply:
Thanks! The notation is indeed easier to read, and it avoids issues with Lean not being able to guess `𝕜`.